### DISCUSSION TITLE
Cahoot (CoinJoin) semi decentralised providing

### DISCUSSION THREAD

**Discussion by cryptoxico** [comit-network/xmr-btc-swap#723] [Ideas]
P2P Semi-Centralised cahoot providing.
The purpose of this protocol is providing Cahoots partners from different parties 

Example : 
Bob want to atomic-swap his bitcoins for Monero.
John want to buy something on JoinMarket

We can have a service installed on theses market/sites/ect they can communicate together with the bitcoin lightning network and organisating cahoots processus

**Comment by thomaseizinger** [comment]
I don't fully follow what exactly you are proposing but it seems to be more of a discussion rather than an actual issue, so I am going to move this to a discussion :)

**Comment by cryptoxico** [comment]
Hi i tought many days on this question and i have a good idea for CoinJoin an atomic swap, if we have a integration of paynym's adress first ? 
PayNym is a protocol for automatically coinjoin your transaction with an special adress used on samouraï wallet

**Reply by thomaseizinger** [reply]
On what level would such an integration be required? Can you perhaps draw up a sequence diagram (using https://www.websequencediagrams.com/ for example) that describes the flow and the involved parties in more detail?

**Reply by cryptoxico** [reply]
Ok imagine you want to buy some monero, of course an atomic swap is uncensorable but everybody knows you swapped your bitcoins for monero so if we have PayNym adress or a CoinJoin compatibility for mixing bitcoins on a transaction

**Reply by thomaseizinger** [reply]
I still don't understand how exactly the integration is meant to work. Can you draw up a diagram on how you imagine this? I am unsure we need first-class support for something like this in the tool or whether we cannot just layer this on top by doing a coin-join either before or after you use the tool.

**Comment by cryptoxico** [comment]
if you need an example for a coinjoin transaction
https://github.com/Samourai-Wallet/Whirlpool

**Reply by binarybaron** [reply]
You could run your coins through whirlpool before depositing them into the swap tool, but that is outside the scope of this project.
